### PR TITLE
[core] improve VAE encode/decode framewise batching

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
# What does this PR do?

Fixes https://github.com/a-r-r-o-w/cogvideox-factory/issues/33#issuecomment-2413487035

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sayakpaul

### CODE REVIEW

File: src/diffusers/models/autoencoders/autoencoder_kl_cogvideox.py [hunk 1]
```diff
@@ -1182,7 +1182,8 @@ def _encode(self, x: torch.Tensor) -> torch.Tensor:
 
         frame_batch_size = self.num_sample_frames_batch_size
         # Note: We expect the number of frames to be either `1` or `frame_batch_size * k` or `frame_batch_size * k + 1` for some k.
-        num_batches = num_frames // frame_batch_size if num_frames > 1 else 1
+        # So, it is okay to not round up as the extra remaining frame is handled in the loop
```
Comment (sayakpaul) [inline src/diffusers/models/autoencoders/autoencoder_kl_cogvideox.py:1185]: Could make this comment more formal and clearer.

File: src/diffusers/models/autoencoders/autoencoder_kl_cogvideox.py [hunk 2]
```diff
@@ -1330,7 +1331,8 @@ def tiled_encode(self, x: torch.Tensor) -> torch.Tensor:
             row = []
             for j in range(0, width, overlap_width):
                 # Note: We expect the number of frames to be either `1` or `frame_batch_size * k` or `frame_batch_size * k + 1` for some k.
-                num_batches = num_frames // frame_batch_size if num_frames > 1 else 1
+                # So, it is okay to not round up as the extra remaining frame is handled in the loop
```
Comment (sayakpaul) [inline src/diffusers/models/autoencoders/autoencoder_kl_cogvideox.py:1334]: Same.

**Review by sayakpaul — APPROVED** [review]
Thanks!

File: src/diffusers/models/autoencoders/autoencoder_kl_cogvideox.py (hunk 1 above)
Comment (a-r-r-o-w) [reply src/diffusers/models/autoencoders/autoencoder_kl_cogvideox.py]: updated, WDYT now?

Comment (sayakpaul) [reply src/diffusers/models/autoencoders/autoencoder_kl_cogvideox.py]: Looks good!

**Pull request merged**